### PR TITLE
[#490] Consolidation of TCA dependencies - 2nd batch

### LINKED ISSUE #490: Consolidation of TCA dependencies - 2nd batch
Continuous work of #477 but due to massive changes it's split into batches.

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #490 

- All dependencies consolidated. 
- Unit and snapshot tests fixed

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

**Review by Chlup — APPROVED** [review]

File: secant.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -599,6 +641,7 @@
 				0D4E7A1926B364180058B01E /* secantTests */,
 				0D4E7A2426B364180058B01E /* secantUITests */,
 				0D4E7A0626B364170058B01E /* Products */,
+				9EB863872922C394003D0F8B /* Recovered References */,
```
Comment (pacu) [inline secant.xcodeproj/project.pbxproj:644]: uhhmm this is always in indication that something can be going on with the project file.

File: secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        static let endpointMainnetAddress = "lightwalletd.electriccoin.co"
+        static let endpointTestnetAddress = "lightwalletd.testnet.electriccoin.co"
+        static let endpointPort = 9067
+        static let mnemonicWordsMaxCount = 24
+        static let requiredTransactionConfirmations = 10
+        static let streamingCallTimeoutInMillis = Int64(10 * 60 * 60 * 1000) // ten hours
+    }
+}
+
+struct ZCashSDKEnvironment {
```
Comment (pacu) [inline secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift:30]:
```suggestion
struct ZcashSDKEnvironment {
```

File: secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import ZcashLightClientKit
+
+extension DependencyValues {
+    var zcashSDKEnvironment: ZCashSDKEnvironment {
+        get { self[ZCashSDKEnvironment.self] }
+        set { self[ZCashSDKEnvironment.self] = newValue }
+    }
+}
+
+extension ZCashSDKEnvironment {
```
Comment (pacu) [inline secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift:18]:
```suggestion
extension ZashSDKEnvironment {
```

**Review by pacu — CHANGES_REQUESTED** [review]

File: secant/Dependencies/DatabaseFiles/DatabaseFilesTestKey.swift [hunk 4]
```diff
@@ -5,18 +5,38 @@
...
 }
+
+extension URL {
+    /// The `DatabaseFilesClient` API returns an instance of the URL or throws an error.
+    /// In order to use placeholders for the URL we need a URL instance, hence `emptyURL` and force unwrapp.
+    static let emptyURL = URL(string: "http://empty.url")!// swiftlint:disable:this force_unwrapping
+}
+
+extension DatabaseFilesClient {
+    static let noop = Self(
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Dependencies/DatabaseFiles/DatabaseFilesTestKey.swift:32]: There are many more instances of `noop` so I refactored all of them.

File: secant/Dependencies/Mnemonic/MnemonicTestKey.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        randomMnemonic: XCTUnimplemented("\(Self.self).randomMnemonic", placeholder: ""),
+        randomMnemonicWords: XCTUnimplemented("\(Self.self).randomMnemonicWords", placeholder: []),
+        toSeed: XCTUnimplemented("\(Self.self).toSeed", placeholder: []),
+        asWords: XCTUnimplemented("\(Self.self).asWords", placeholder: []),
+        isValid: XCTUnimplemented("\(Self.self).isValid")
+    )
+}
+
+extension MnemonicClient {
+    static let noop = Self(
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Dependencies/Mnemonic/MnemonicTestKey.swift:22]: There are many more instances of `noop` so I refactored all of them.

File: secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift (hunk 2 above)
Comment (LukasKorba) [reply secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift:30]: Refactored all `ZCashSDKEnvironment` instances.

File: secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift (hunk 3 above)
Comment (LukasKorba) [reply secant/Dependencies/ZCashSDKEnvironment/ZCashSDKEnvironmentInterface.swift:18]: Refactored all ZCashSDKEnvironment instances.

File: secant.xcodeproj/project.pbxproj (hunk 1 above)
Comment (LukasKorba) [reply secant.xcodeproj/project.pbxproj:644]: resolved

File: secant/Dependencies/SDKSynchronizer/SDKSynchronizerInterface.swift [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    case enhancing
+    case fetching
+    case minedTransaction
+    case foundTransactions
+    case failed
+    case connectionStateChanged
+}
+
+protocol SDKSynchronizerClient {
+    var blockProcessor: CompactBlockProcessor? { get }
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Dependencies/SDKSynchronizer/SDKSynchronizerInterface.swift]: `blockProcessor` removed from the protocol.

**Review by Chlup — APPROVED** [review]

**Review by pacu — APPROVED** [review]

**Pull request merged**